coralproject/talk
# Upgrade `akismet-api` and use new payload types for spam checks

Special thanks to @Sentynel for bringing this to our attention in their fork PR (https://github.com/coralproject/talk/pull/4123)! We used the suggestions from their PR to update our Akismet types and submission style for our spam filters!

## What does this PR do?

- Updates our `akismet-api` dependency
- Uses new types for the payloads sent through the `AkismetClient` to check for spam

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- Configure the spam tooling (Akismet) under `Admin > Configure > Moderation > Spam detection filter`
    - Make sure the filter is turned on (i.e. `Spam detection filter` option)
    - Be sure to fill out both the api key and site url
    - Try using both `IP-based spam detection` in both on and off modes
- Put a debug break point into: `server/src/core/server/services/comments/pipeline/phases/spam.ts` at Line 111
- Post comments
    - Can try using some spam comment examples from around the internet if you'd like
- See it processing the `isSpam` state for the comments

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.
